File: eventbus/eclipse_cpp/eventbus/src/unittest/UnitTest.cpp
#include <boost/thread.hpp>
#include <EventEmitterExample.hpp>
#include <LoggerExample.hpp>

static bool shutdown = false;

void sigint_handler(int sig) {

   shutdown = true;
}

int main(int argc, char *argv[]) {

   signal(SIGINT, sigint_handler);

   EBRouter *router = new EBRouter("", 5);
   EventEmitterExample* eventEmitter1 = new EventEmitterExample(router, 1);
   EventEmitterExample* eventEmitter2 = new EventEmitterExample(router, 2);
   LoggerExample* logger = new LoggerExample();

   router->addSubscriber("logger", logger);
   router->startReq();

   eventEmitter1->start();
   eventEmitter2->start();

   while (true) {

      if (logger->hasEvents())
         logger->processEvents();

      if (shutdown) {

         router->shutdownReq();

         while (!router->threadIsComplete())
            boost::this_thread::sleep(boost::posix_time::milliseconds(200));

         abort();
      }

      boost::this_thread::sleep(boost::posix_time::milliseconds(1000));
   }

   return 0;
}

File: eventbus/eclipse_java/eventbus/src/orbisoftware/msgblit/eventbus/unittest/UnitTest.java
package orbisoftware.msgblit.eventbus.unittest;

import orbisoftware.msgblit.eventbus.EBRouter;

public class UnitTest {

   public static void main(String args[]) {

      EBRouter router = new EBRouter("", 30);
      EventEmitterExample eventEmitter1 = new EventEmitterExample(router, 1);
      EventEmitterExample eventEmitter2 = new EventEmitterExample(router, 2);
      LoggerExample logger = new LoggerExample();
      
      router.addSubscriber("logger", logger);
      router.start();

      eventEmitter1.start();
      eventEmitter2.start();

      while (true) {

         if (logger.hasEvents())
            logger.processEvents();

         try {
            Thread.sleep(1000);
         } catch (InterruptedException e) {
         }
      }
   }
}
